### PR TITLE
[ringbuffer] Write msgs to file with parquet and support dumping all messages in file

### PR DESCRIPTION
**Pull request by RanMaoyi**
Still need more UT and test

### CODE REVIEW

File: src/supplemental/nanolib/ringbuffer/ringbuffer.c [hunk 1]
```diff
@@ -200,52 +201,171 @@ static inline int ringBuffer_rule_check(ringBuffer_t *rb, void *data, int flag)
 	return 0;
 }
 
-static int write_msgs_to_file(ringBuffer_t *rb)
+void ringbuffer_parquet_cb(void *arg)
 {
-	int ret = 0;
```
Comment (wanghaEMQ) [inline src/supplemental/nanolib/ringbuffer/ringbuffer.c:205]: Check nng_aio_result() first

File: src/supplemental/nanolib/ringbuffer/ringbuffer.c [hunk 2]
```diff
@@ -200,52 +201,171 @@ static inline int ringBuffer_rule_check(ringBuffer_t *rb, void *data, int flag)
...
+		keys[i] = rb->msgs[i].key;
+		smsgs[i] = rb->msgs[i].data;
+		darray[i] = nng_msg_payload_ptr((nng_msg *)rb->msgs[i].data);
+		dsize[i] = nng_msg_len((nng_msg *)rb->msgs[i].data) -
+		        (nng_msg_payload_ptr((nng_msg *)rb->msgs[i].data) -
+				(uint8_t *)nng_msg_body((nng_msg *)rb->msgs[i].data));
+	}
+
+	nng_aio *aio;
+	nng_aio_alloc(&aio, ringbuffer_parquet_cb, file);
```
Comment (wanghaEMQ) [inline src/supplemental/nanolib/ringbuffer/ringbuffer.c:307]: so where to free?

File: src/supplemental/nanolib/ringbuffer/ringbuffer.c (hunk 1 above)
Comment (JaylinYu) [reply src/supplemental/nanolib/ringbuffer/ringbuffer.c:205]: solid review

Comment (RanMaoyi) [reply src/supplemental/nanolib/ringbuffer/ringbuffer.c:205]: fixed

File: src/supplemental/nanolib/ringbuffer/ringbuffer.c (hunk 2 above)
Comment (RanMaoyi) [reply src/supplemental/nanolib/ringbuffer/ringbuffer.c:313]: That's a question to think about, and I'll think about it more tomorrow

**Review by lee-emqx — APPROVED** [review]

**Review by JaylinYu — APPROVED** [review]
well done

**Pull request merged**